src/language-services/formatter.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as PQF from "@microsoft/powerquery-formatter";
+import * as PQP from "@microsoft/powerquery-parser";
+
+import { FormattingOptions, Range, TextDocument, TextEdit } from "vscode-languageserver-types";
+
+export function tryFormatconnection(
+    document: TextDocument,
+    formattingOptions: FormattingOptions,
+    localizationTemplates: PQP.ILocalizationTemplates,
+): PQP.Result<TextEdit[], string> {
```

Comment:
Can we return something that isn't parser specific?

Reply:
Fixed.